shm/Cargo.cpp

```diff
@@ -0,0 +1,19 @@
+#include "Cargo.hpp"
+
+Cargo& Cargo::operator+=(const size_t amount) {
+    amount_ += amount;
+    return *this;
+}
+
+Cargo& Cargo::operator-=(const size_t amount) {
+    if (amount > amount_) {
+        amount_ = 0;
+        return *this;
```

Comment:
So if you want to decrease bigger cargo amount than you have you are selling everything, nice :)